Packs/ServiceNowv2/Integrations/ServiceNowv2/ServiceNowv2.py

```diff
@@ -0,0 +1,1542 @@
+
+    if fields:
+        for field in fields:
+            body[field] = fields[field]
+
+    if custom_fields:
+        for field in custom_fields:
+            # custom fields begin with "u_"
+            if field.startswith('u_'):
+                body[field] = custom_fields[field]
+            else:
+                body['u_' + field] = custom_fields[field]
+
+    return body
+
+
+def split_fields(fields: str) -> dict:
+    """Split str fields of Demisto arguments to SNOW request fields.
+
+    Args:
+        fields: fields in a string representation.
+
+    Returns:
+        dic_fields object for SNOW requests.
+    """
+    dic_fields = {}
+
+    if fields:
+        # As received by the command
+        arr_fields = fields.split(';')
+
+        for f in arr_fields:
+            field = f.split('=')
+            if len(field) > 1:
+                dic_fields[field[0]] = field[1]
+
+    return dic_fields
+
+
+class Client(BaseClient):
+    """
+    Client to use in the ServiceNow integration. Overrides BaseClient.
+    """
+
+    def __init__(self, server_url: str, username: str, password: str, verify: bool, proxy: bool, fetch_time: str,
+                 sysparm_query: str, sysparm_limit: int, timestamp_field: str, ticket_type: str, get_attachments: bool):
+        """
+
+        Args:
+            server_url: SNOW serveer url
+            username: SNOW username
+            password: SNOW password
+            verify: whether to verify the request
+            proxy: whether to allow proxy
+            fetch_time: first time fetch for fetch_incidents
+            sysparm_query: system query
+            sysparm_limit: system limit
+            timestamp_field: timestamp field for fetch_incidents
+            ticket_type: default ticket type
+            get_attachments: whether to get ticket attachments by default
+        """
+        self._base_url = server_url
+        self._verify = verify
+        self._username = username
+        self._password = password
+        self._proxies = handle_proxy() if proxy else None
+        self.fetch_time = fetch_time
+        self.timestamp_field = timestamp_field
+        self.ticket_type = ticket_type
+        self.get_attachments = get_attachments
+        self.sys_param_query = sysparm_query
+        self.sys_param_limit = sysparm_limit
+        self.sys_param_offset = 10
+
+    def send_request(self, path: str, method: str = 'get', body: dict = None, params: dict = None,
+                     headers: dict = None, file=None):
+        """
+        Generic request to ServiceNow.
+        """
+        body = body if body is not None else {}
+        params = params if params is not None else {}
+
+        url = f'{self._base_url}{path}'
+        if not headers:
+            headers = {
+                'Accept': 'application/json',
+                'Content-Type': 'application/json'
+            }
+        if file:
+            # Not supported in v2
+            url = url.replace('v2', 'v1')
```

Comment:
what dose this do?

Reply:
if you want to upload a file to an incident/ticket you can not do it using the v2 of the API. so we're changing it.
https://docs.servicenow.com/bundle/geneva-servicenow-platform/page/integrate/inbound_rest/reference/r_AttachmentAPI-POSTmultipart.html
@liorblob correct me if needed.